src/main/java/org/interledger/InterledgerPacket.java

```diff
@@ -0,0 +1,41 @@
+package org.interledger;
+
+import org.interledger.ilp.InterledgerAddress;
+
+/**
+ * <p>Interledger Packets are attached to Interledger transfers to facilitate Interledger
+ * payments.</p>
+ *
+ * @param <T> The generic type of this packet, as defined by implementations.
+ * @param <D> The generic type of the data payload, as defined by implementations.
+ */
+public interface InterledgerPacket<T extends InterledgerPacketType, D> {
```

Comment:
Per @earizon's comment in the main "discussion" area, this should probably be as follows:

```
public interface InterledgerPacket<T extends InterledgerPacketType, D> {
 T getType();
 D getEncodedData();
}
```

The methods `getDestinationAmount` and `getDestinationAccount` should be moved down into `InterledgerPaymentPacket.java`.